xds/internal/xdsclient/clientimpl_watchers.go

```diff
@@ -45,7 +45,9 @@ func (c *clientImpl) WatchResource(rType xdsresource.Type, resourceName string,
 
 	if err := c.resourceTypes.maybeRegister(rType); err != nil {
 		logger.Warningf("Watch registered for name %q of type %q which is already registered", rType.TypeName(), resourceName)
-		c.serializer.TrySchedule(func(context.Context) { watcher.OnError(err, func() {}) })
+		c.serializer.TrySchedule(func(context.Context) {
+			watcher.OnError(fmt.Errorf("[xDS node id: %v]: %v", c.nodeID, err), func() {})
```

Comment:
Wondering if we should do something so that `watcher.OnError` always prefixes the node id automatically?

```go
watcher := c.wrapWatcher(watcher)
```

or something?

Reply:
Thanks for the idea. This definitely reduces the amount of change in this PR.